devito/symbolics/extended_sympy.py

```diff
@@ -312,6 +304,7 @@ def eval_bhaskara_sin(expr):
 
 
 def eval_taylor_sin(expr):
+    from devito.finite_differences import Mul
```

Comment:
to avoid all these local imports, I wonder whether we should leave Add/Mul/ etc in this module, but have them inherit from our own ``finite_difference.py``'s Add/Mul, rather than sympy's. I don't see why the Add/Mul/.. in ``finite_difference.py`` should inherit from ``FrozenExpr`` either

Reply:
>I wonder whether we should leave Add/Mul/ etc in this module
Wouldn't it be a bit odd and confusing?

> I don't see why the Add/Mul/.. in finite_difference.py should inherit from FrozenExpr either

The way it was before